Backend/core/data_sources/serial_source.py

```
@@ -0,0 +1,182 @@
+                            timestamp = parsed_data.get('tstamp_unix', 0)
+                            last_data_time = time.time()
+                            self.is_connected = True
+                            self._emit_data(parsed_data, timestamp)
+                else:
+                    time.sleep(0.1)
+                
+                # Check for timeout (5 seconds without data)
+                if time.time() - last_data_time > 5:
+                    self.is_connected = False
+                    
+            except Exception as e:
```

Ruff (0.14.13)

Do not catch blind exception: `Exception`

(BLE001)